srtp/srtp.c

```diff
@@ -233,6 +233,60 @@ srtp_stream_alloc(srtp_stream_ctx_t **str_ptr,
    return stat;    
   }
 
+  if (p->enc_xtn_hdr && p->enc_xtn_hdr_count > 0) {
+    srtp_cipher_type_id_t enc_xtn_hdr_cipher_type;
+    int enc_xtn_hdr_cipher_key_len;
+
+    str->enc_xtn_hdr = (int*) srtp_crypto_alloc(p->enc_xtn_hdr_count * sizeof(p->enc_xtn_hdr[0]));
+    if (!str->enc_xtn_hdr) {
+      auth_dealloc(str->rtcp_auth);
+      srtp_cipher_dealloc(str->rtcp_cipher);
+      auth_dealloc(str->rtp_auth);
+      srtp_cipher_dealloc(str->rtp_cipher);
+      srtp_crypto_free(str->limit);
+      srtp_crypto_free(str);
+      return srtp_err_status_alloc_fail;
+    }
+    memcpy(str->enc_xtn_hdr, p->enc_xtn_hdr, p->enc_xtn_hdr_count * sizeof(p->enc_xtn_hdr[0]));
+    str->enc_xtn_hdr_count = p->enc_xtn_hdr_count;
+
+    /* For GCM ciphers, the corresponding ICM cipher is used for header extensions encryption. */
+    switch (p->rtp.cipher_type) {
+    case SRTP_AES_128_GCM:
+      enc_xtn_hdr_cipher_type = SRTP_AES_128_ICM;
+      enc_xtn_hdr_cipher_key_len = 30;
```

Comment:
Both constants are in `crypto/include/aes_icm_ossl.h` which is only pulled in to this file if `OPENSSL` and `OPENSSL_KDF` are defined.